os/arch/arm/src/armv7-a/arm_gicv2.c

```diff
@@ -508,6 +508,92 @@ int up_prioritize_irq(int irq, int priority)
 	return -EINVAL;
 }
 
+/****************************************************************************
+ * Name: up_set_irq_affinity
+ *
+ * Description:
+ *   set the affinity of a given irq (SPI only) to given cpu set
+ *
+ * Input Parameters:
+ *   irq: irq id
+ *   cpuset: cpu set
+ *   Note: the API will only support 8 bit cpu set.
+ *
+ * Returned Value:
+ *   OK : operation successful
+ *   EINVAL: irq out of range
+ *
+ ****************************************************************************/
+
+int up_set_irq_affinity(int irq, cpu_set_t cpu_set)
+{
+	DEBUGASSERT(irq >= GIC_IRQ_SPI && irq < NR_IRQS);
+	DEBUGASSERT(sizeof(cpu_set) == sizeof(uint8_t));
+
+	/* Ignore invalid interrupt IDs */
+
+	if (irq >= GIC_IRQ_SPI && irq < NR_IRQS) {
+
+		/* read the current cpu affinity register */
+		uint32_t reg_value = getreg32(GIC_ICDIPTR(irq));
+
+		/* clear the bits for the current irq */
+		reg_value &= ~(0xFF << ((irq % 4) * 8));
+
+		/* write the target core for the irq */
+		reg_value |= (cpu_set << ((irq % 4) * 8));
```

Comment:
Please use macros for this, 

https://github.com/apache/nuttx/blob/48b93b8dc807c99a8903827d20f23eb02286ad67/arch/arm/src/armv7-a/arm_gicv2.c#L615-L632